go/vt/vtgate/engine/delete.go

```diff
@@ -145,6 +145,24 @@ func (del *Delete) RouteType() string {
 	return delName[del.Opcode]
 }
 
+// KeyspaceTableNames specifies the table that this primitive routes to
+func (del *Delete) KeyspaceTableNames() []*KeyspaceTableName {
+	if del.Table != nil {
```

Comment:
Again I think this kind of thing would be simpler and less repetitive if the Keyspace and Table were returned separately.